pkg/yang/entry.go

```diff
@@ -95,9 +95,11 @@ type Entry struct {
 	// is a module only.
 	Identities []*Identity `json:",omitempty"`
 
-	Augments   []*Entry                   `json:"-"` // Augments associated with this entry.
+	Augments   []*Entry                   `json:"-"` // Augments defined in this entry.
+	Augmented  []*Entry                   `json:"-"` // Augments merged into this entry.
 	Deviations []*DeviatedEntry           `json:"-"` // Deviations associated with this entry.
 	Deviate    map[deviationType][]*Entry `json:"-"`
+	Uses       map[*Uses]*Entry           `json:"-"` // Uses merged into this entry
```

Comment:
Looks like JSON only supports the `string` type as map keys. What do you think is a good solution to encode the `map[*Uses]*Entry` map? Should I encode `*Uses `as its field `Name`, which is type `string` and represents the name of the used `grouping`?

Reply:
One option here would be to change to make this something like:

```golang
type UsesStmt struct {
  Uses *Uses
  Grouping *Entry
}

type Entry struct {
  Uses []*UsesStmt
}
```

this would remove the need for the `map` here, to replace it with a slice. The downside of doing this is that we can't look up based on a `Uses` any more (however, this seems unlikely that one would do this, since it needs a pointer to the actual `Entry` that we don't store).

I think its important that we can persist the whole `Uses` since it gives us a bunch of additional information rather than just the grouping name.

Reply:
That's a good idea. I will give it a try.

Reply:
Ok, I have converted Uses to []*UsesStmt and update marshal_test.go. Please let me know what you think!